### PR TITLE
Chore: ESlint import order

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

This adds a new eslint rule to require imports be in a consistent order. Apart from just being "nice", this makes it easier to read the imports (the order is what you expect, no surprises) and easier to add to them later, which becomes important as more and more people are working with the codebase.

I think it's also a good way to identifying (and managing) weird import-order dependencies, or imports that have side effects. If it's required that your imports are declared in a certain order, that probably points to _another_ problem that should be fixed.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

I am so, so sorry to literally everyone in the org getting pinged for a review 😅 

If you want to read through exactly(ish) what I've done in this PR, the commits are a good place to start - I broke them up into logical chunks of config, auto-fix, and then manually fixes.

### CODE REVIEW

**Comment by dprokop** [comment]
Think I'm game - I have the same, dunno why by I like those imoprts sorted.

**Comment by zoltanbedi** [comment]
I'm also supporting this! 🥇

**Comment by joshhunt** [comment]
Hmmm... tests are failing (I think) because we have a circular dependency within the prometheus datasource.

```
                 prometheus/querybuilder/binaryScalarOperations.ts
depends on       prometheus/querybuilder/shared/operationUtils.ts
which depends on prometheus/querybuilder/shared/operationUtils.ts
which depends on prometheus/querybuilder/components/LabelParamEditor.tsx
which depends on prometheus/datasource.tsx
which depends on prometheus/add_label_to_query.ts
which depends on prometheus/querybuilder/parsing.ts
which depends on prometheus/querybuilder/binaryScalarOperations.ts 😱
```

I'm digging into it further.

File: public/app/plugins/datasource/prometheus/querybuilder/binaryScalarOperations.ts [hunk 1]
```diff
@@ -71,6 +71,14 @@ export const binaryScalarDefs = [
   },
 ];
 
+export const binaryScalarOperatorToOperatorName = binaryScalarDefs.reduce((acc, def) => {
```
Comment (joshhunt) [inline public/app/plugins/datasource/prometheus/querybuilder/binaryScalarOperations.ts:74]: @grafana/observability-metrics @aocenas Specifically, could you take a look over the changes I made in [this commit](https://github.com/grafana/grafana/pull/44959/commits/afbaeb3edb1dd69574272a8f31a9d3b57511d5dc) please to resolve an issue caused by circular dependencies.

Note: the circular dependencies are still there, I just mitigated an issue caused by it.

**Comment by joshhunt** [comment]
All tests are passing, but _I think_ because I changed so many files I've retriggered a bunch of CodeQL warnings. I think I'm find to ignore these in this PR, but ill have to defer to someone else.

**Review by jackw — APPROVED** [review]
LGTM! 🚀 Amazing work on this @joshhunt 👏 

Have left a comment regarding the change to the eslintconfig. There's a chance we may need to revert that.

File: .eslintignore [hunk 2]
```diff
@@ -20,3 +20,6 @@ vendor
 # Auto-generated localisation files
 public/locales/_build/
 public/locales/**/*.js
+
+# Autogenerated by github.com/grafana/icons
```
[the remark this answers is not in the document]
Comment (joshhunt) [reply .eslintignore:24]: lol really? Would be useful to know what's going on here 😅 

I'm happy to just ignore linting for this one file for now, but then get a grasp on whether this file actually is automatically linted or not.

[the remark this answers is not in the document]
Comment (jackw) [reply .eslintignore]: Certainly would be good to know. I agree with just ignoring it for now and get this merged. 👍

**Comment by joshhunt** [comment]
I expected this backport failure. Will backport manually.

**Pull request merged**